src/features/profile/helpers/TrustedDevices/useDeviceInfos.ts

```diff
@@ -1,34 +1,34 @@
 import { useEffect, useState } from 'react'
+// eslint-disable-next-line no-restricted-imports
+import { browserName } from 'react-device-detect'
 import DeviceInfo from 'react-native-device-info'
 
-import { getDeviceModelFromUserAgent } from 'features/profile/helpers/TrustedDevices/getDeviceModelFromUserAgent'
 import { getUniqueId } from 'libs/react-native-device-info/getUniqueId'
 
 export interface DeviceInformations {
   deviceId: string
-  deviceOS: string
-  deviceModel: string
+  os?: string
+  source?: string
 }
 
 export const useDeviceInfos = (): DeviceInformations => {
```

Comment:
Je m'interroge sur l'intérêt d'un hook plutôt qu'un singleton...
@bruno-ebstein-pass-culture ou @dbenfouzari vous avez un avis ?

Reply:
Effectivement aucune utilité à être un hook de mon point de vue.
Ca ferait l'affaire ça je pense.

```suggestion
export async function getDeviceInfos(): DeviceInformations () {
  const deviceId = getUniqueId()
  const osWeb = await DeviceInfo.getBaseOs()
  const osNative = DeviceInfo.getSystemName()
  const modelNative = DeviceInfo.getModel()
  
  return {
    deviceId,
    os: osNative !== 'unknown' ? osNative : osWeb,
    source: modelNative !== 'unknown' ? modelNative : browserName,
  }
}
```

En constante (du genre `const deviceInfos = {}` ce n'est pas possible puisqu'on a besoin de `await` certaines choses, mais de cette façon ça devrait le faire.

Reply:
C'est pas faux... Dans ce cas je le verrai splitté et avec un "vrai" intérêt 

```ts
export async function getDeviceInfos(): DeviceInformations {
  const deviceId = getUniqueId()
  const osWeb = await DeviceInfo.getBaseOs()
  const osNative = DeviceInfo.getSystemName()
  const modelNative = DeviceInfo.getModel()
  
  return {
    deviceId,
    os: osNative !== 'unknown' ? osNative : osWeb,
    source: modelNative !== 'unknown' ? modelNative : browserName,
  }
}

export function useDeviceInfos() {
  const [isLoading, setIsLoading] = useState(true);
  const [data, setData] = useState();

  useEffect(() => {
    setIsLoading(true)

    (async() => {
      setData(await getDeviceInfos())
    })()

    setIsLoading(false)
  }, [])

  return { isLoading, data }:
}
```

Reply:
j'ai l'impression qu'on pourrait simplifier de cette manière

```ts
async function getDeviceInfos(): Promise<DeviceInformations> {
  const deviceId = getUniqueId()
  const osWeb = await DeviceInfo.getBaseOs()
  const osNative = DeviceInfo.getSystemName()
  const modelNative = DeviceInfo.getModel()
  
  return {
    deviceId,
    os: osNative !== 'unknown' ? osNative : osWeb,
    source: modelNative !== 'unknown' ? modelNative : browserName,
  }
}

export function useDeviceInfos(): DeviceInformations | undefined {
  const [data, setData] = useState<DeviceInformations | undefined>();

  useEffect(() => {
    (async() => {
      setData(await getDeviceInfos())
    })()
  }, [])

  return data
}
```

Reply:
Oui j'ai écrit à l'arrache sans taper dans mon IDE donc forcément c'estp as tout bon :P
Ceci dit le `DeviceInformations | undefined` n'est pas nécessaire je crois. Mais bon, c'est du détail

Reply:
Je ne trouve pas ça très lisible / utile de créer deux fonctions dans ce cas.. :/

Reply:
Sauf que tu peux tester plus facilement ta fonction getDeviceInfos sans passer par un hook ;)

Reply:
C'est toujours bien de séparer ce qu'on peut. Ca permet de tester plus facilement, de mocker plus facilement, de rendre le code plus évolutif, etc etc.

Bon là j'avoue dans ce cas précis bof mais bon, c'est par principe.

Reply:
Pour la tester il faudrait exporter `getDeviceInfos()` et j'avoue que l'exporter juste pour la tester je ne suis pas fan. 
En plus quelqu'un pourrait potentiellement l'utiliser ailleurs et sans utiliser le hook `useDeviceInfos()`